woocommerce/woocommerce-android
# Refactored customer info card section in Order Detail

This PR takes care of the third step of the Order Detail refactoring #2844 .

#### Changes
- Adds a new model `Address` that can be reused in both `Order` and `ShippingLabel` features.
- Adds the Order Customer Info card (`OrderDetailCustomerInfoView`) to the Order detail screen.
- Removes most of the methods in `AddressUtils` since they were moved to the `Address` model.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- This PR just adds the Order detail warning card to the Order detail screen. The remaining cards will be added in subsequent PRs along with unit tests and click actions for all the buttons in the Order detail.
**- This PR is in draft till this [Step 2 PR](https://github.com/woocommerce/woocommerce-android/pull/2857) can be merged.**

#### Screenshots
##### Virtual Products - Hide shipping section:
<img src="https://user-images.githubusercontent.com/22608780/93559305-09557a80-f99d-11ea-9ecb-41c0e460c03b.png" width="300"/>
##### Physical products - No shipping address:
<img src="https://user-images.githubusercontent.com/22608780/93559333-1e320e00-f99d-11ea-8836-992cd99d7ff6.png" width="300"/>
##### Physical products - All information available:
<img src="https://user-images.githubusercontent.com/22608780/93559365-3013b100-f99d-11ea-8fba-c3cb0dc00c02.png" width="300"/>
##### No billing address available - display only billing phone and email:
<img src="https://user-images.githubusercontent.com/22608780/93559393-41f55400-f99d-11ea-82aa-5c1279db2455.png" width="300"/>
##### No billing phone available - display only billing address and email:
<img src="https://user-images.githubusercontent.com/22608780/93559422-52a5ca00-f99d-11ea-9201-c6add9513b54.png" width="300"/>
##### No billing email available - display only billing address and phone:
<img src="https://user-images.githubusercontent.com/22608780/93559444-605b4f80-f99d-11ea-92da-58c55945557f.png" width="300"/>
##### No billing information available - display only shipping section:
<img src="https://user-images.githubusercontent.com/22608780/93559474-736e1f80-f99d-11ea-8259-1db12c9ad71f.png" width="300"/>


#### Testing
- Click on an order with only physical products. Verify that the shipping and billing section is displayed correctly. 
- Click on an order with only virtual products. Verify that the billing section is displayed correctly. 
- Click on an order without any billing information. Verify that the shipping section is displayed correctly. 
- Click on an order without any billing phone. Verify that the billing email and address is displayed correctly. 
- Click on an order without any billing email. Verify that the billing phone and address is displayed correctly. 
- Click on an order without any billing address. Verify that the billing phone and email is displayed correctly. 

**Note that the click actions in this card will take place in another PR**

Update release notes:
